packages/order-utils/src/order_factory.ts

```diff
@@ -1,17 +1,17 @@
-import { ECSignature, SignedOrder } from '@0xproject/types';
+import { ECSignature, Order, SignedOrder } from '@0xproject/types';
 import { BigNumber } from '@0xproject/utils';
 import { Provider } from 'ethereum-types';
 import * as ethUtil from 'ethereumjs-util';
 import * as _ from 'lodash';
 
+import { constants } from './constants';
 import { orderHashUtils } from './order_hash';
 import { generatePseudoRandomSalt } from './salt';
 import { ecSignOrderHashAsync } from './signature_utils';
 import { MessagePrefixType } from './types';
 
 export const orderFactory = {
-    async createSignedOrderAsync(
-        provider: Provider,
+    createOrder(
```

Comment:
This function doesn't feel like it's doing much to me. Why is `expirationTimeSecondsIfExists` the only optional parameter? You can definitely make a case for default `takerAddress`, `senderAddress`, `feeRecipientAddress`, and `salt` values, so only making `expirationTimeSeconds` optional feels random to me. IMO this function would be more useful if it took a partial order as input and returned a full order with sensible defaults.

Reply:
I can make more of them optional, this PR just refactored the existing one slightly so that we can create a normal order w/o the signature